sparkmagic/sparkmagic/kernels/kernelmagics.py

```diff
@@ -447,6 +447,17 @@ def get_session_settings(self, line, force):
             else:
                 return None
 
+    def _initialize_auth(self, args):
+        full_class = conf.authenticators().get(args.auth, conf.authenticators().get("None"))
+        module, class_name = (full_class).rsplit('.', 1)
+        events_handler_module = importlib.import_module(module)
+        auth_instance = getattr(events_handler_module, class_name)
+        #fix this
```

Comment:
Is this still relevant?

Reply:
Yes. In kernelmagics.py, it is letting the user specify username, password, url, and auth inline. But it takes the auth as a String. So this function initiates the auth class instance from string they passed in. And if it is Basic it will set the username and password they specify. Not quite sure the difference between kernelmagics.py vs remotesparkmagics.py in both I have this initialize auth class to convert the auth string into an instance.